lib/proof/data/data_sources/circuits_download_data_source.dart

```diff
@@ -6,103 +6,70 @@ import 'package:http/http.dart' as http;
 
 import 'package:path_provider/path_provider.dart';
 import 'package:polygonid_flutter_sdk/proof/domain/entities/download_info_entity.dart';
+import 'package:polygonid_flutter_sdk/proof/domain/entities/download_response_entity.dart';
 import 'package:polygonid_flutter_sdk/sdk/di/injector.dart';
 
 class CircuitsDownloadDataSource {
-  ///
-  Future<bool> circuitsFilesExist() async {
-    String fileName = 'circuits.zip';
-    Directory directory = await getApplicationDocumentsDirectory();
-    String path = directory.path;
-    var file = File('$path/$fileName');
-    return await file.exists();
-  }
+  StreamController<DownloadResponseEntity> _controller =
+      StreamController<DownloadResponseEntity>();
 
-  ///
-  Future<String> getPathToCircuitZipFile() async {
-    Directory directory = await getApplicationDocumentsDirectory();
-    String path = directory.path;
-    String fileName = 'circuits.zip';
-    return '$path/$fileName';
-  }
+  Stream<DownloadResponseEntity> get downloadStream => _controller.stream;
 
-  ///
-  Future<String> getPathToCircuitZipFileTemp() async {
-    Directory directory = await getApplicationDocumentsDirectory();
-    String path = directory.path;
-    String fileName = 'circuits_temp.zip';
-    return '$path/$fileName';
-  }
+  int _downloadSize = 0;
 
-  ///
-  Future<String> getPath() async {
-    Directory directory = await getApplicationDocumentsDirectory();
-    String path = directory.path;
-    return path;
-  }
+  late http.Client _client;
+
+  /// downloadSize
+  int get downloadSize => _downloadSize;
 
   ///
-  Future<http.StreamedResponse> getStreamedResponseFromServer() async {
+  Future<void> initStreamedResponseFromServer() async {
+    http.Client client = http.Client();
+
     const bucketUrl =
         "https://circuits.polygonid.me/circuits/v1.0.0/polygonid-keys.zip";
 
     var request = http.Request('GET', Uri.parse(bucketUrl));
 
-    final http.StreamedResponse response = await http.Client().send(request);
-    return response;
-  }
-
-  ///
-  Future<void> deleteFile(String pathToFile) async {
-    try {
-      var file = File(pathToFile);
-      await file.delete();
-    } catch (_) {
-      // file not found? no problem, we don't need it
-    }
-  }
-
-  ///
-  void renameFile(String pathTofile, String newPathToFile) {
-    var file = File(pathTofile);
-    file.renameSync(newPathToFile);
-  }
+    final http.StreamedResponse response = await client.send(request);
+    _downloadSize = response.contentLength ?? 0;
 
-  ///
-  void writeZipFile({
-    required String pathToFile,
-    required List<int> zipBytes,
-  }) {
-    var file = File(pathToFile);
-    file.writeAsBytesSync(
-      zipBytes,
-      mode: FileMode.append,
+    response.stream.listen(
+      (List<int> newBytes) {
+        _controller.add(
+          DownloadResponseEntity(
+            newBytes: newBytes,
+          ),
+        );
+      },
+      onDone: () {
+        _controller.add(DownloadResponseEntity(
```

Comment:
The DS doesn't build the entity, it's done either by the UC or a mapper used on the response of the DS by the repo.